src/main/scala/io/iohk/ethereum/consensus/pow/validators/OmmersValidator.scala

```diff
@@ -30,11 +30,16 @@ trait OmmersValidator {
   )(implicit blockchainConfig: BlockchainConfig): Either[OmmersError, OmmersValid] = {
 
     val getBlockHeaderByHash: ByteString => Option[BlockHeader] = blockchainReader.getBlockHeaderByHash
-    val bestBranch = blockchainReader.getBestBranch()
     val getNBlocksBack: (ByteString, Int) => List[Block] =
       (_, n) =>
```

Comment:
Actually, the first parameter should not be ignored, especially now that we are using hashes in the blocks getter function.
The end result should be the same, as the call the getNBBlockBack() are passing the parentHash as parameter.

Proposal pseudo code:

```
    val getNBlocksBack: (ByteString, Int) => List[Block] =
      (tailBlockHash, n) =>
        Iterator
          .iterate(blockchainReader.getBlockByHash(tailBlockHash))(
            _.flatMap(block => blockchainReader.getBlockByHash(block.header.parentHash))
          )
          .take(n)
          .collect { case Some(block) => block }
          .toList
          .reverse

```

Reply:
Indeed. If the result is the same, what do you think about removing the parameter as it essentially complexify the interface needlessly ?

Reply:
It depends if one let the lambda provider or the caller have control of the chain range.

However, it seems that `BlockValidation.getNBlocksBackFromChainOrQueue()` implements this interface, and is actually using the hash parameter, so we may have to keep it that way, and make every implementing function uses its parameters.